src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,107 @@
+import classNames from 'classnames';
+import React from 'react';
+import { removeTodo } from '../../api/todos';
+import { Error, Filter, Todo } from '../../types/Todo';
+
+type Props = {
+  setFilter: (filter: Filter) => void;
+  filter: Filter;
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+  setHasError: (value: Error) => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  setFilter,
+  filter,
+  todos,
+  setTodos,
+  setHasError,
+}) => {
+  const countItems = todos.filter(todo => !todo.completed).length;
+
+  const filterAllHandler = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+
+    setFilter(Filter.All);
+  };
+
+  const filterActiveHandler = (e: React.MouseEvent<HTMLAnchorElement>) => {
```

Comment:
It is not really needed to move each filter to separate function
If you want separate func you can pass argument instead